draft-ietf-tsvwg-rfc4960-bis.xml

```diff
@@ -1021,9 +1021,10 @@ parameters and has to report them according to
 <xref target='sec_parameter_format'/>, it SHOULD bundle the ERROR chunk
 containing the "Unrecognized Parameters" error cause with the chunk sent
 in response (e.g., COOKIE ECHO).
-If the receiver of the INIT ACK chunk cannot bundle the COOKIE ECHO chunk with
-the ERROR chunk, the ERROR chunk MAY be sent separately but not before the
-COOKIE ACK chunk has been received.</t>
+If the receiver of the chunk with unrecognized parameters cannot bundle the
+ERROR chunk with the chunk sent in response,
+the ERROR chunk MAY be sent separately but not before the
+chunk sent in response has been received.</t>
```

Comment:
I think your suggestion misses the point we want to make:

In the special case of unrecognized parameters in the INIT ACK chunk and the inability of bundling the COOKIE-ECHO chunk with the ERROR chunk, the ERROR chunk can be sent separately, but only after the COOKIE ACK has been received. This avoids the case that if the packet containing the ERROR chunk arrives before the packet containing the COOKIE ECHO chunk arrives. If that would happen, the ERROR would result in destroying the association.

In other situations, there should not be a problem with arriving the packet with the ERROR chunk first. Therefore I think this change is not needed.

Reply:
I'm marking this as resolved. If you don't agree, please unresolve this conversation.

Reply:
This was trying to be editorial consistency -- in the previous sentence we say "the chunk sent in response (e.g., COOKIE ECHO)".  To me, that suggests that "the chunk sent in response" is the preferred identifier for that chunk, and "e.g., COOKIE ECHO" indicates that COOKIE ECHO is a typical type of chunk to be sent in response but that other types of chunk are possible.  Your first response suggests that the COOKIE ECHO nature is quite important here, and points out that I had conflated COOKIE ECHO and COOKIE ACK in part of my suggested change.  So, the COOKIE ACK text needs to stay as COOKIE ACK, I definitely agree.  Additionally, if the only possible "chunk sent in response" is COOKIE ECHO, my proposal would be to s/e.g./i.e./ in the previous sentence and continue to use COOKIE ECHO (and COOKIE ACK) in this sentence.  If other chunk types are possible, then my proposal would be to apply s/the/a/ in a few places (and continue to use COOKIE ECHO and COOKIE ACK), since the use of the definite article implies that there is always exactly one in the situation being described.

Reply:
`If the receiver of any other chunk` refers to any chunk different from the INIT chunk. In the  context of this base specification, the INIT ACK chunk is the canonical example, but there are other chunks (mostly defined in extensions) this text applies to.

The text in the sentence you are improving proves a consideration for a special case, which has two prerequisites:
1. The chunk containing unrecognized parameters is an INIT ACK chunk and
2. the receiver of the INIT ACK chunk can not bundle the COOKIE ECHO with the ERROR chunk

Only if these two conditions are met, the sending of the ERROR needs to be delayed until the COOKIE ACK has been received.

How about using:
```
If the receiver of an INIT ACK chunk cannot bundle the COOKIE ECHO chunk with
the ERROR chunk, the ERROR chunk MAY be sent separately but not before the
COOKIE ACK chunk has been received.</t>
```
Does that address your issue?